src/mono/mono/mini/interp/jiterpreter.c

```diff
@@ -743,7 +746,11 @@ trace_info_get (gint32 index) {
 
 static gint32
 trace_info_alloc () {
+#ifdef DISABLE_THREADS
 	gint32 index = trace_count++,
+#else
+	gint32 index = atomic_fetch_add ((atomic_int *)&trace_count, 1),
```

Comment:
I wanted to make sure that all of the atomics in jiterpreter.c use the same API instead of mixing the two types of atomics, since the semantics are a little different